src/iap.ts

```diff
@@ -513,23 +555,24 @@ const App = () => {
 
  */
 
-export const requestPurchase = ({
-  sku,
-  andDangerouslyFinishTransactionAutomaticallyIOS = false,
-  obfuscatedAccountIdAndroid,
-  obfuscatedProfileIdAndroid,
-  appAccountToken,
-  skus, // Android Billing V5
-  isOfferPersonalized = undefined, // Android Billing V5
-  quantity,
-  withOffer,
-}: RequestPurchase): Promise<ProductPurchase | void> =>
+export const requestPurchase = (
+  request: RequestPurchase,
+): Promise<ProductPurchase | void> =>
   (
     Platform.select({
       ios: async () => {
-        if (!sku) {
-          return Promise.reject(new Error('sku is required for iOS purchase'));
+        if (!('sku' in request)) {
+          throw new Error('sku is required for iOS purchase');
```

Comment:
I replaced `return Promise.reject` with `throw`, since those are equivalent in an async function but throw is a bit safer, since we can occasionally forget to include `return`.